pkg/constants/constants.go

```diff
@@ -114,6 +114,11 @@ const (
 	ArgumentPredictorHost = "--predictor_host"
 )
 
+// InferenceService container name
+const (
+	InferenceServiceContainerName = "kfserving-container"
```

Comment:
Hmm -- should we instead name this after the Component name? I.e. Predictor.

Reply:
sure, we can do it, but it will make logic more complex but will not introducing any obvious benefit.
for example, we want to check a Predictor Pod information, its name already has `Predictor` (for example `flowers-sample-predictor-default-vh8vg-deployment-74d59c7bwmkwc`), we can know the Pod is for Predictor, then we get its yaml definition, I think it is enough to recognize "kfserving-container" container.

Reply:
Noo what I mean is just use the string value of this: https://github.com/kubeflow/kfserving/blob/master/pkg/constants/constants.go#L94

Reply:
for now explainer also need the container name to mount volume, I think we can just use a unified name

Reply:
I'm thinking ahead to when we might want to host multiple containers in the same pod for performance reasons. We might not even be using knative (and just raw deployments).

It should be pretty trivial to simply name the container based off of the component that it is (i.e. predictor/explainer/transformer).

Does that make sense?

Reply:
This PR introduces specified container name to solve container recognition in webhook for knative constraints which cannot support mount volume. 
if in future we only use deployment, we can just define the volume mount in deployment definition directly. Then I think introducing container name is no meaning then.

Reply:
Sure.